### ISSUE TITLE
Make Tensorflow 2.1 available through conda

### ISSUE THREAD

**Issue by cossio** [tensorflow/tensorflow#35754]
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 18.04
- TensorFlow installed from (source or binary): conda
- Python version: 3.7

**Describe the problem**

`conda install tensorflow` installs version 2.0 instead of 2.1.
Please, make tensorflow 2.1 available through conda.

**Comment by mihaimaruseac** [comment]
I think conda support is community driven, not provided by us. Adding @gunan to confirm/infirm

**Comment by cossio** [comment]
Maybe there is another repo where this issue can be raised?

**Comment by gunan** [comment]
@jjhelmus may be able to help.
We can redirect conda availability issues to Jonathan, I think?

**Comment by jjhelmus** [comment]
The team at Anaconda is looking to make Tensorflow 2.1 packages available.  This process may take a few weeks.

**Comment by cossio** [comment]
Having `pip` and `conda` together is a pain. Why tensorflow is released on pip by default?

**Comment by mihaimaruseac** [comment]
Umm, because that's where the python packages are?

**Comment by cossio** [comment]
Sorry, I meant why is a conda release not automatic on the pipeline? PyTorch has a conda channel where the package appears very soon after release.

**Comment by mihaimaruseac** [comment]
We don't have infrastructure for that and I don't think it is prioritized

**Comment by wuliytTaotao** [comment]
I prefer to install tensorflow with conda, because conda does not require us to consider GPU support. Hope tensorflow 2.1 can be installed through conda as soon as possible.

**Comment by edwardyehuang** [comment]
Frameworks like Pytorch even have their own Conda channel. Shipping the latest version via Conda can bring the easiest installation experience for our users (So we don't need consider about the GPU enviorment setup etc.). In Conda, we only need to "conda install tensorflow-gpu", everything will be setup and ready to use.

**Comment by byronyi** [comment]
Community driven means you could setup a conda channel to distribute your own TensorFlow packages. It’s an open source project with permissive license, so feel free to do whatever you’d like with it.

**Comment by woodknight** [comment]
> We don't have infrastructure for that and I don't think it is prioritized

Maybe that's one of many reasons why PyTorch is more and more popular than TensorFlow.

**Comment by gavinlive** [comment]
From https://www.anaconda.com/tensorflow-in-anaconda/:

"Anaconda is proud of our efforts to deliver a simpler, faster experience using the excellent TensorFlow library. It takes significant time and effort to add support for the many platforms used in production, and to ensure that the accelerated code is still stable and mathematically correct. As a result, **our TensorFlow packages may not be available concurrently with the official TensorFlow wheels**. We are, however, committed to maintaining our TensorFlow packages, and work to have updates available as soon as we can."

**Comment by cossio** [comment]
> Maybe that's one of many reasons why PyTorch is more and more popular than TensorFlow.

Ease of installing is definitely a plus.

**Comment by joanafilipa** [comment]
@woodknight @cossio please stay on topic as per community guidelines. Thank you!

**Comment by jackd** [comment]
@joanafilipa which part of those comments isn't on topic? It's all well and good for @mihaimaruseac to say this isn't a priority, but a community shouldn't be discouraged from expressing frustration when those priorities push them away. Judging by reactions to posts - which certainly isn't an exhaustive survey, but better than nothing - @cossio isn't the only one who feels this way.

**Comment by pgunn** [comment]
At risk of continuing a conversation that's already off-topic,
Please recall that expressing frustration is not productive and it's something that a lot of us who subscribe to the issue just looking for updates get emails about. It's okay to *be* frustrated, but please do it somewhere else so the github issues are not full of chatter.

**Comment by edwardyehuang** [comment]
Agree. However, the result of the "productive " comment is 

> We don't have infrastructure for that and I don't think it is prioritized.



Frustrated or not, I think those users only want the Tensorflow become better.

Also, I believe the Pytorch Conda is supported by offical [1, 2]. We did not off-topic. We are only requesting the offical Tensorflow to support the Conda distribution without rely too much on the community.


> Maybe that's one of many reasons why PyTorch is more and more popular than TensorFlow.

Unfortunately, this is the truth, especially in Research [3]. And for my personal experience, most of researchers around me are moved to Pytorch.

I understand the Tensorflow Developers are working very hard to deliver the new features and fix the bugs. Thank you very much. 
However, installation of Tensorflow enviorment is the most important things since this is the entrance to use Tensorflow. And I can tell you lots of Researchers or Users have zero enginner or developer background, which means they only know how to design the algorithm, but they can very easy in trouble with installation.


[1] https://anaconda.org/pytorch
[2] https://pytorch.org/get-started/locally/
[3] https://towardsdatascience.com/is-pytorch-catching-tensorflow-ca88f9128304

**Comment by byronyi** [comment]
@jjhelmus Anything we could help to provide future TF releases on Anaconda in a more timely manner? 

TF 2.2 is upcoming and @edwardyehuang @cossio you might considering to join the effort to make the release on conda faster.

**Comment by theadactyl** [comment]
Thanks @edwardyehuang for the context. 

We hear that this is a point of frustration for conda users. @gunan could add the appropriate label (feature? or build/install?) to this so we can better track this issue going forward? We'll circle round with the TF and community folks who add conda support and see what the blockers are to 
 getting packages ready sooner after new TF releases.

A general reminder to keep things constructive and on-topic. If you have additional data points from the TF ecosystem you'd like to mention, please do.

**Comment by FeVo96** [comment]
I think Tensorflow 2.1.0 is no available on Anaconda.

I reinstalled it today and installed tensorflow, here it automatically installed Tf 2.1.0

My steps:
Open anaconda prompt
-> conda create -n tf Tensorflow // here it downloaded Tensorflow 2.1.0 

I dont know if there is the possibility to update it by any Command. But this way works.

**Comment by wuliytTaotao** [comment]
> I think Tensorflow 2.1.0 is now available on Anaconda.
> 
> I reinstalled it today and installed tensorflow, here it automatically installed Tf 2.1.0
> 
> My steps:
> Open anaconda prompt
> -> conda create -n tf Tensorflow // here it downloaded Tensorflow 2.1.0
> 
> I dont know if there is the possibility to update it by any Command. But this way works.

@FeVo96 Is your reinstalled TF2.1 GPU version?

I just looked at https://repo.anaconda.com/pkgs/main/linux-64/ and found tensorflow-2.1 CPU version, but no GPU version.

**Comment by FeVo96** [comment]
> > I think Tensorflow 2.1.0 is now available on Anaconda.
> > I reinstalled it today and installed tensorflow, here it automatically installed Tf 2.1.0
> > My steps:
> > Open anaconda prompt
> > -> conda create -n tf Tensorflow // here it downloaded Tensorflow 2.1.0
> > I dont know if there is the possibility to update it by any Command. But this way works.
> 
> @FeVo96 Is your reinstalled TF2.1 GPU version?
> 
> I just looked at https://repo.anaconda.com/pkgs/main/linux-64/ and found tensorflow-2.1 CPU version, but no GPU version.

It is also the gpu version available.
I just opened up the conda environment in the prompt and typed conda Upgrade tensorflow-gpu. And it works

**Comment by edwardyehuang** [comment]
> 
> 
> > > I think Tensorflow 2.1.0 is now available on Anaconda.
> > > I reinstalled it today and installed tensorflow, here it automatically installed Tf 2.1.0
> > > My steps:
> > > Open anaconda prompt
> > > -> conda create -n tf Tensorflow // here it downloaded Tensorflow 2.1.0
> > > I dont know if there is the possibility to update it by any Command. But this way works.
> > 
> > 
> > @FeVo96 Is your reinstalled TF2.1 GPU version?
> > I just looked at https://repo.anaconda.com/pkgs/main/linux-64/ and found tensorflow-2.1 CPU version, but no GPU version.
> 
> It is also the gpu version available.
> I just opened up the conda environment in the prompt and typed conda Upgrade tensorflow-gpu. And it works

Tensorflow 2.1 should included GPU

**Comment by PRDrum5** [comment]
tensorflow 2.1.0 and tensorflow-gpu 2.1.0 both of windows

**Comment by cossio** [comment]
I don't think you need a separate gpu package for 2.1. Can someone confirm?

**Comment by JoranMichiels** [comment]
I can confirm that conda's tensorflow 2.1 defaults to CPU (and with tensorflow-gpu 2.0 the same code ran on GPU). But the 'official' tensorflow 2.1 (not tensorflow-gpu) [release](https://github.com/tensorflow/tensorflow/releases/tag/v2.1.0) does state that, from now on, GPU is always included and is default...

**Comment by html04** [comment]
For windows: if you conda install tensorflow-gpu==2.1.0 you get the current version.

**Comment by JoranMichiels** [comment]
Conda's tensorflow-gpu 2.1.0 has also become available for Linux. Issue can be closed, I guess.

**Comment by mihaimaruseac** [comment]
Closing per request

**Comment by larsmathuseck** [comment]
What about the macOS version?

**Comment by sanchit88** [comment]
Any update regarding conda and Tensorflow 2.1 on macOS? It is still installing 2.0

**Comment by timpitman** [comment]
I'd also like to see Mac support for TensorFlow 2.1 - I've filed https://github.com/ContinuumIO/anaconda-issues/issues/11697 since this issue has been closed.

**Comment by byronyi** [comment]
There is no CUDA support for macOS anyway, so why don't you guys simply do `pip3 install tensorflow==2.1.0`?

**Comment by acegilz** [comment]
pip3 install tensorflow==2.1.0
ERROR: Could not find a version that satisfies the requirement tensorflow==2.1.0 (from versions: none)
ERROR: No matching distribution found for tensorflow==2.1.0

**Comment by seanpmorgan** [comment]
@acegilz  , You likely need to update pip
```
pip3 install --upgrade pip
pip3 install tensorflow==2.1.0
```

**Comment by acegilz** [comment]
pip3 install --upgrade pip
Requirement already up-to-date: pip in /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages (20.0.2)

pip3 install tensorflow==2.1.0
ERROR: Could not find a version that satisfies the requirement tensorflow==2.1.0 (from versions: none)
ERROR: No matching distribution found for tensorflow==2.1.0

**Comment by byronyi** [comment]
@acegilz TensorFlow 2.1 did not publish Py3.8 package on macOS. Luckily we got 2.2 soon and it should come with Py3.8 support.

**Comment by mihaimaruseac** [comment]
@acelitz: Py3.8 support #33374

You can always check the pypi package page to see which pips are available.

**Comment by cciprianmihai** [comment]
@cossio I am trying to install `tf 2.x` via `conda`, but I am receiving a different error for both versions: `tf 2.0.0` and `tf 2.1.0`.

I saw that there is another possible way of solving this, by installing `tf 2.1.0` using `pip` as:
```
conda create --name tf2
pip install tensorflow
```

Is this a proper way of doing it?

Thank you!

**Comment by wuliytTaotao** [comment]
@cciprianmihai Use `conda install` instead of `pip install`, and you should set the version of TF, like `conda install tensorflow-gpu=2.1` for TF 2.1 GPU version.

**Comment by coded5282** [comment]
Will tensorflow-2.2.0 be on conda soon as well? Thanks.

**Comment by mihaimaruseac** [comment]
We haven't yet released the final 2.2 pips. Once that happens, @jjhelmus can trigger the conda builds, afaik

**Comment by fcakyon** [comment]
@jjhelmus any eta on the tensorflow-gpu 2.2.0 conda build? Thanks.

**Comment by jjhelmus** [comment]
Typically, it takes anywhere from 2 to 4 weeks for us (Anaconda, Inc) to build Tensorflow conda packages.  I expect that the 2.2.0 release will take a similar amount of time.

**Comment by fcakyon** [comment]
Thanks a lot for the feedback!

**Comment by jeffheaton** [comment]
Any updates on TF 2.2?

**Comment by gunan** [comment]
Please file conda related issue through anaconda issue tracker, located at:
https://github.com/ContinuumIO/anaconda-issues/issues

Anaconda manages the TF releases on their platform, and they do not monitor this issue tracker.

**Comment by MadMan1981** [comment]
Any updates on TensorFlow >2.1 for Conda users on Linux and macOS ? thanks.

**Comment by edwardyehuang** [comment]
> 
> 
> Any updates on TensorFlow >2.1 for Conda users on Linux and macOS ? thanks.

I think Tensorflow 2.2 has already been releated at conda long time ago.
[https://anaconda.org/anaconda/tensorflow](https://anaconda.org/anaconda/tensorflow)

**Comment by kmalfroid** [comment]
macOS is still stuck at 2.0.0:
https://anaconda.org/anaconda/tensorflow
<img width="202" alt="Screen Shot 2021-02-12 at 10 15 22" src="https://user-images.githubusercontent.com/73581880/107749654-396e3300-6d1b-11eb-85be-02a0472dc40a.png">

Any update on this?

**Comment by byronyi** [comment]
You should really direct this question to anaconda.
